src/repl/scala/tools/nsc/interpreter/InteractiveReader.scala

```diff
@@ -46,6 +46,13 @@ object InteractiveReader {
     }
 
   def apply(): InteractiveReader = SimpleReader()
+
+  // a non-interactive SimpleReader that returns the given text
+  def apply(text: String): InteractiveReader = SimpleReader(
```

Comment:
Maybe. `apply` gives you a `SimpleReader` for some reason. `apply(text)` gives you the same thing, but it only reads from that input. As a user, I don't care that it's a `SimpleReader`. WDYT? If I were closer to the city, I'd buy you a beer to discuss.